api/internal/features/container/controller/protected.go

```diff
@@ -0,0 +1,26 @@
+package controller
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/raghavyuva/nixopus-api/internal/features/logger"
+	shared_types "github.com/raghavyuva/nixopus-api/internal/types"
+)
+
+func (c *ContainerController) isProtectedContainer(containerID string, action string) (*shared_types.Response, bool) {
```

Comment:
this can go in init file rather than a seperate file because it does not require an api endpoint if i'm right,